File: utils/KeywordProvider.js
class KeywordProvider{
    
    //検索用キーワードを作成
    static getKeyword(artist,titile){
        let keyword="";
        for(let word of [artist,titile]){
        word = word.toLowerCase();
        const startIndex = word.indexOf("feat");
        word = startIndex !== -1 ? word.substring(0, startIndex) : word;
        keyword += " " +word;
        }
        return keyword.trim();
    }
}

module.exports = KeywordProvider;

File: app/utils/KeywordProvider.py
def get_keyword(artist: str, title: str) -> str:
    keyword = ""
    for word in [artist, title]:
        word = word.lower()
        start_index = word.find("feat")
        word = word[:start_index] if start_index != -1 else word
        keyword += " " + word
    return keyword.strip()
